ethereum/extethclient/eth_wallet_client.go

```diff
@@ -301,62 +317,136 @@ func (c *ethClient) Raw() *ethclient.Client {
 	return c.ec
 }
 
-func (c *ethClient) CancelTxWithNonce(
+// Transfer transfers ETH to the given address, nonce Lock()/Unlock() handling
+// is done internally. The gasLimit parameter is required when transferring to a
+// contract and ignored otherwise.
+func (c *ethClient) Transfer(
 	ctx context.Context,
-	nonce uint64,
-	gasPrice *big.Int,
-) (ethcommon.Hash, error) {
+	to ethcommon.Address,
+	amount *coins.WeiAmount,
+	gasLimit *uint64,
+) (*ethtypes.Receipt, error) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
```

Comment:
The previous version of `Transfer` was grabbing the nonce-synchronization lock, but then it wasn't waiting for a confirmation, defeating the purpose of the lock. You wouldn't know if the transfer succeeded or not and even if it did succeed, it wouldn't show up in the balance while the TX was still in the mempool.  The new version still grabs the lock, but it also waits for the receipt or error. The previous version of `CancelTxWithNonce` was grabbing the nonce-synchronization lock, and that was removed. With a fixed nonce, there is no nonce to synchronize, and attempting to grab the lock is not what we want if a different method, that we wait to fail, already has the lock and is blocked trying to get a receipt.